bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java

```diff
@@ -1265,6 +1257,60 @@ private static Iterable<HistoricItem> getAllStatesBetween(Item item, ZonedDateTi
         }
     }
 
+    private static Iterable<HistoricItem> getAllStatesBetweenWithStartingState(Item item, ZonedDateTime begin,
```

Comment:
Ok, sounds good. I like the idea of reusing the code for `PersistenceResource`. 
What do you think about creating a new function `getAllStatesBetweenWithBoundaries` instead of extending the `getAllStatesBetween` with the flag? Of course, it would use the `getAllStatesBetween` and just add the boundaries, as it is currently in `getAllStatesBetweenWithStartingState`. It  would replace the `getAllStatesBetweenWithStartingState` and could be used in both `PersistenceResource` and `PersistenceExtensions`.

Reply:
I'm not totally against a new method, but where you see the benefit for that (compared to a parameter)?

Reply:
I see mainly two benefits:
1. The methods are shorter and simpler.
2. The code is easier to understand. For example: The call `getAllStatesBetweenWithBoundaries(...)` is more informative than `getAllStatesBetween(..., true)`. In the latter case, you must always check the name of the parameter to understand its meaning.

Reply:
Can you do the necessary changed to `getAllStatesBetweenWithBoundaries`? Otherwise I think we are good to merge.